redash/models.py

```diff
@@ -493,8 +493,7 @@ def to_dict(self, all=False, with_permissions_for=None):
                 DataSourceGroup.group == with_permissions_for,
                 DataSourceGroup.data_source == self).one()[0]
 
-        doc_url = self.options.get('doc_url', schema['properties'].get(
-            'doc_url', {}).get('default'))
+        doc_url = self.options.get('doc_url')
```

Comment:
Are you sure the issue was with this line and not [the code below](https://github.com/alison985/redash/blob/f26976caed6f5a5fabb7bcf137f9437a78d0b595/redash/models.py#L498)?

```python
if doc_url:
   d['options'] = {'doc_url': doc_url}
```

It was overriding `options` with just `{'doc_url': doc_url}`, while what it was supposed to be doing is probably `d['options'].update(doc_url=doc_url)`. Your change might have fixed it as a side effect, if `doc_url` was empty when testing.